velociraptor/catalogue/soap_catalogue.py

```diff
@@ -0,0 +1,266 @@
+
+    def print_fields(self):
+        print("SOAP catalogue fields used:")
+        for name in self.names_used:
+            print(f"  {name}")
+
+    def _register_quantities(self):
+        with h5py.File(self.file_name, "r") as handle:
+            self.root = CatalogueGroup(self.file_name, "", handle)
+            cosmology = handle["SWIFT/Cosmology"].attrs
+            self.units = SimpleNamespace()
+            self.a = cosmology["Scale-factor"][0]
+            self.units.scale_factor = cosmology["Scale-factor"][0]
+            self.z = cosmology["Redshift"][0]
+            self.units.redshift = cosmology["Redshift"][0]
+            H0 = cosmology["H0 [internal units]"][0]
+            Omega_m = cosmology["Omega_m"][0]
+            Omega_lambda = cosmology["Omega_lambda"][0]
+            w0 = cosmology["w_0"][0]
+            Omega_b = cosmology["Omega_b"][0]
+            if w0 != -1.0:
+                self.cosmology = wCDM(
+                    H0=H0, Om0=Omega_m, Ode0=Omega_DE, w0=w_of_DE, Ob0=Omega_b
+                )
+            else:
+                # No EoS
+                self.cosmology = FlatLambdaCDM(H0=H0, Om0=Omega_m, Ob0=Omega_b)
+            try:
+                boxsize = handle["SWIFT/Header"].attrs["BoxSize"][0]
+            except:
+                boxsize = 1000.0
+            boxsize *= unyt.Mpc
+            physical_boxsize = self.a * boxsize
+            self.units.box_length = boxsize
+            self.units.comoving_box_volume = boxsize ** 3
+            self.units.period = physical_boxsize
+            self.units.physical_box_volume = physical_boxsize ** 3
+            self.units.cosmology = self.cosmology
+
+    def get_SOAP_quantity(self, quantity_name):
+        path = []
+        for path_part in quantity_name.split("."):
+            # attribute names cannot start with a number
+            if path_part[0].isnumeric():
+                path.append(f"v{path_part}")
+            else:
+                path.append(path_part)
+        value = reduce(getattr, path, self.root)
+        self.names_used.add(quantity_name)
+        return value
+
+    def get_quantity(self, quantity_name):
+        try:
+            return super().get_quantity(quantity_name)
+        except AttributeError:
+            pass
+        try:
+            return self.get_SOAP_quantity(quantity_name)
+        except AttributeError:
+            pass
+        try:
+            SOAP_quantity_name, colidx = VR_to_SOAP(quantity_name)
+            quantity = self.get_SOAP_quantity(SOAP_quantity_name)
+            if colidx >= 0:
+                return quantity[:, colidx]
+            else:
+                return quantity
+        except NotImplementedError as err:
+            if quantity_name in [
+                "apertures.veldisp_star_10_kpc",
+                "apertures.veldisp_star_30_kpc",
+            ]:
+                print(f"Ignoring missing {quantity_name}...")
+                return None
+            raise err
```

Comment:
What is going on here? Why this special case? Please document and explain. Print statements for these things are generally a bad idea, too. You should raise a Warning, maybe?